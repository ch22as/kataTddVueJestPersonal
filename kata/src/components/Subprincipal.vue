<template>
  <div class="container">
      <h1>Subprincipal</h1>
      <img src="https://picsum.photos/200/300" alt="">
  </div>
</template>

<script>
export default {
    name: 'Subprincipal'

}
</script>

<style>

</style>