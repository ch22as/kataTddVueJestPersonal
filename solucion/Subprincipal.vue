<template>
  <div class="container subprincipal">
    <h1>Subprincipal</h1>
    <img src="https://picsum.photos/200/300" alt="Imagen random" />
    <Worker/>
  </div>
</template>

<script>
import Worker from './Worker.vue';

export default {
    name: 'Subprincipal',
    components: {
        Worker
    }

}
</script>

<style>
.subprincipal {
  background: blue;
  padding: 1%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
</style>