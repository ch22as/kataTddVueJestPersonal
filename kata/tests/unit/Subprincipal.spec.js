import { mount } from '@vue/test-utils';
import Subprincipal from '@/components/Subprincipal';

const build = () =>{
    const wrapper = mount(Subprincipal);

    return {
        wrapper,
        h1: () => wrapper.find('h1'),
        img: () => wrapper.find('img')
    }
}

describe('Subprincipal', () =>{
    it('it renders the component?', () =>{
        const { wrapper } = build();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('it render right?', () =>{
        const { h1, img} = build();

        expect(h1().exists()).toBe(true);
        expect(img().exists()).toBe(true);
    })
})