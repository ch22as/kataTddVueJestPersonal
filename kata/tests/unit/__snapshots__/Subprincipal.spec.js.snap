// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subprincipal it renders the component? 1`] = `
<div class="container">
  <h1>Subprincipal</h1> <img src="https://picsum.photos/200/300" alt="">
</div>
`;
